src/main/java/com/khomishchak/cryptoportfolio/controllers/ExchangerController.java

```diff
@@ -46,22 +48,25 @@ public List<Balance> getAccountBalances(@RequestAttribute long userId) {
         return exchangerService.getAllMainBalances(userId);
     }
 
-    @DeleteMapping("/{exchangerCode}/balance")
-    public DeleteExchangerResp deleteAccountBalance(@RequestAttribute long userId, @PathVariable String exchangerCode) {
-        return exchangerService.deleteExchangerForUser(userId, ExchangerCode.valueOf(exchangerCode));
+    @DeleteMapping("/{balanceId}/balance")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteAccountBalance(@PathVariable long balanceId) {
```

Comment:
Why do you have account's balance related logic inside exchanger controller. From perspective of reading this controller should provide API to manage exchanger resource. Please explain how does an exchanger as an entity handle account's balance and what is a relation between them.

Reply:
this controller is responsible for everything that is related to the 3rd party exchangers, and this balance is a part of exact exchanger(each exchanger integration have its own balance), and so balance is isolated exchanger resource

Reply:
Path variable means a path to current resource. In your case it's `Exchanger` resource so path variable should be some identifier of exchanger, e.g id, unique name etc. If exchanger has list of balances then it should looks like:
```suggestion
    @DeleteMapping("/balances/{balanceId}")
```

Reply:
for now it is not required change. For now one exchanger for one user can have only one balance

Reply:
@mkhomishchak If I got it there is no `Exchanger` entity itself. The representation if it is a `Balance` entity and each account may have 0 or more balances. If it's why isn't better to have a user's API what handles users' resources. Then it will be more intuitive `/users/{userId}/balances/{balanceId} `